_posts/pl/2021-04-26-zarzadzanie-dostepem-przy-uzyciu-access-control-list.md

```diff
@@ -0,0 +1,324 @@
+
+W naszej przykładowej aplikacji dodajemy dane bezpośrednio do bazy danych za pomocą DML. Kwestią otwartą i nieporuszną w tym artykule jest zaimplementowanie serwisu, który jest w stanie programowo zarządzać uprawnieniami.
+
+```sql
+-- Kilka przykładowych zadań
+
+INSERT INTO tasks (id, chapter, title, description, creation_date) VALUES (1, 'Security', 'tytuł 1', 'opis zadania 1', current_timestamp);
+INSERT INTO tasks (id, chapter, title, description, creation_date) VALUES (2, 'Cloud', 'tytuł 2', 'opis zadania 2', current_timestamp);
+INSERT INTO tasks (id, chapter, title, description, creation_date) VALUES (3, 'Security', 'tytuł 3', 'opis zadania 3', current_timestamp);
+INSERT INTO tasks (id, chapter, title, description, creation_date) VALUES (4, 'Cloud', 'tytuł 4', 'opis zadania 4', current_timestamp);
+INSERT INTO tasks (id, chapter, title, description, creation_date) VALUES (5, 'Frontend', 'tytuł 5', 'opis zadania 5', current_timestamp);
+INSERT INTO tasks (id, chapter, title, description, creation_date) VALUES (6, 'Security', 'tytuł 6', 'opis zadania 6', current_timestamp);
+INSERT INTO tasks (id, chapter, title, description, creation_date) VALUES (7, 'Cloud', 'tytuł 7', 'opis zadania 7', current_timestamp);
+INSERT INTO tasks (id, chapter, title, description, creation_date) VALUES (8, 'Storage', 'tytuł 8', 'opis zadania 8', current_timestamp);
+
+-- ACL - przepisanie uprawnień
+
+INSERT INTO acl_sid (id, principal, sid) VALUES
+(1, 0, 'ROLE_TASK'),
+(2, 1, 'user1'),
+(3, 1, 'user2');
+
+INSERT INTO acl_class (id, class) VALUES
+(1, 'com.consdata.task.model.Task');
+
+INSERT INTO acl_object_identity (id, object_id_class, object_id_identity, parent_object, owner_sid, entries_inheriting) VALUES
+(1, 1, 1, NULL, 2, 0),
+(2, 1, 3, NULL, 2, 0),
+(3, 1, 6, NULL, 2, 0),
+(4, 1, 2, NULL, 3, 0),
+(5, 1, 4, NULL, 3, 0),
+(6, 1, 5, NULL, 3, 0),
+(7, 1, 7, NULL, 3, 0),
+(8, 1, 8, NULL, 3, 0);
+
+INSERT INTO acl_entry (id, acl_object_identity, ace_order, sid, mask, granting, audit_success, audit_failure) VALUES
+(1, 1, 1, 2, 1, 1, 1, 0),
+(2, 2, 1, 2, 1, 1, 1, 0),
+(3, 3, 1, 2, 1, 1, 1, 0),
+(4, 4, 1, 3, 1, 1, 1, 0),
+(5, 5, 1, 3, 1, 1, 1, 0),
+(6, 6, 1, 3, 1, 1, 1, 0),
+(7, 7, 1, 3, 1, 1, 1, 0),
+(8, 8, 1, 3, 1, 1, 1, 0);
+```
+
+Opis poszczególnych tabel został przedstawiony [tutaj](https://docs.spring.io/spring-security/site/docs/current/reference/html5/#domain-acls-key-concepts).
+
+## ACL w akcji
+
+Omówiliśmy już wszystkie niezbędne kwestie. Jesteśmy gotowi uruchomić aplikację i ją przetestować. Pełny kod znajduję się pod adresem [**https://github.com/pawelwalaszek/spring-security-acl**](https://github.com/pawelwalaszek/spring-security-acl).
+
+```
+mvn spring-boot:run
+```
+
+Wejście pod adres:
+
+```
+http://localhost:8080/tasks/list-with-acl
```

Comment:
Czy nie lepiej żeby to wylądowało w linku a nie bloku kodu?

Reply:
Poprawione